### PR TITLE
Add iif (immediate if) template function/filter

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adds documentation for the immediate if template filter/method (`iif`).

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/61428
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -297,6 +297,63 @@ The same thing can also be expressed as a filter:
...
+  Yes
+{% else %}
+  No
+{% endif %}
+```
+
+{% endraw %}
+
+Or using a shorter syntax:
+
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown:320]:
```suggestion
```

File: source/_docs/configuration/templating.markdown [hunk 2]
```diff
@@ -297,6 +297,63 @@ The same thing can also be expressed as a filter:
...
+{% endraw %}
+
+Additionally, to the above, you can use the `iif` function/filter, which is
+an immediate if.
+
+Syntax: `iif(value, if_true, if_false, if_none)`
+
+It expects a value and can return a value if the condition returns true,
+false or `None`. The value is loosely compared if it is truthy or not. For
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown:336]: > The value is loosely compared if it is truthy or not.

What does this mean?

File: source/_docs/configuration/templating.markdown [hunk 3]
```diff
@@ -297,6 +297,63 @@ The same thing can also be expressed as a filter:
...
+{{ is_state('light.kitchen', 'on') | iif('Yes', 'No') }}
+
+{{ (state('light.kitchen') == 'on') | iif('Yes', 'No') }}
+```
+
+{% endraw %}
+
+If the `if_none` parameter is used to define the return value
+of when `value` is `None`. If `if_none` is omitted, the `if_false` value is
+used when the value is `None`.
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]:
```suggestion
```

File: source/_docs/configuration/templating.markdown [hunk 4]
```diff
@@ -297,6 +297,63 @@ The same thing can also be expressed as a filter:
...
+{% endraw %}
+
+Additionally, to the above, you can use the `iif` function/filter, which is
+an immediate if.
+
+Syntax: `iif(value, if_true, if_false, if_none)`
+
+It expects a value and can return a value if the condition returns true,
+false or `None`. The value is loosely compared if it is truthy or not. For
+example, an empty string, mapping or list, will be considered false.
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]:
```suggestion
`iif` returns the value of `if_true` if the condition is truthy, the value of `if_false` if it's `falsy` and the value of `if_none` if it's `None`.
An empty string, an empty mapping or an an empty list, are all falsy, refer to [the Python documentation](https://docs.python.org/3/library/stdtypes.html#truth-value-testing) for an in depth explanation.

`if_true` is optional, if it's omitted `True` is returned if the condition is truthy.
`if_false` is optional, if it's omitted `False` is returned if the condition is falsy.
`if_none` is optional, if it's omitted the value of `if_false` is returned if the condition is `None`.
```

File: source/_docs/configuration/templating.markdown [hunk 5]
```diff
@@ -297,6 +297,63 @@ The same thing can also be expressed as a filter:
...
+```text
+{{ 'Yes' if is_state('light.kitchen', 'on') else 'No' }}
+```
+
+{% endraw %}
+
+Additionally, to the above, you can use the `iif` function/filter, which is
+an immediate if.
+
+Syntax: `iif(value, if_true, if_false, if_none)`
```
Comment (emontnemery) [inline source/_docs/configuration/templating.markdown]:
```suggestion
Syntax: `iif(condition, if_true, if_false, if_none)`
```

**Comment by frenck** [comment]
Thanks for the suggestions, much better. Applied them 👍

**Review by emontnemery — APPROVED** [review]
Thanks, @frenck 👍

**Pull request merged**